server/app/services/applications/ApplicationService.java

```diff
@@ -0,0 +1,54 @@
+package services.applications;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+import javax.inject.Inject;
+import models.Application;
+import play.libs.concurrent.HttpExecutionContext;
+import repository.ApplicationRepository;
+import services.program.ProgramDefinition;
+
+/** The service responsible for mediating access to the Application resource. */
+public final class ApplicationService {
+
+  private final ApplicationRepository applicationRepository;
+  private final HttpExecutionContext httpExecutionContext;
+
+  @Inject
+  ApplicationService(
+      ApplicationRepository applicationRepository, HttpExecutionContext httpExecutionContext) {
+    this.applicationRepository = checkNotNull(applicationRepository);
+    this.httpExecutionContext = checkNotNull(httpExecutionContext);
+  }
+
+  /** Validates the given application is associated with the given program. */
```

Comment:
**question (blocking):** why are we validating that the application is associated with a given program?

Reply:
https://github.com/civiform/civiform/pull/3180

Reply:
"Preventing program admins from accessing applications that aren't part of the provided program" (https://github.com/civiform/civiform/pull/3180)

I separated out the validation logic for the applicant/TI client scenario and the program admin scenario